### PR TITLE
test: add Solana E2E performance tests and improve Solana outbounds performance

### PR DESCRIPTION
**Pull request by skosito**
# Description

contains fixes from this PR to improve outbounds performance: https://github.com/zeta-chain/node/pull/3295 which should probably be backported (cc: @lumtis)

- similar to how stress eth tests work
- for eth there are not erc20 stress tests, here adding tests for both sol and spl - this seems more granural and stable
- using 50 for each of those 4

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Release Notes

- **New Features**
  - Added a `/systemtime` telemetry endpoint in the `zetaclient`.
  - Introduced Solana performance tests for deposits and withdrawals.
  
- **Bug Fixes**
  - Resolved issues with unsupported Solana transaction versions.
  - Enhanced handling of pending nonces during transaction aborts.

- **Tests**
  - Added multiple stress tests for Solana and SPL deposits and withdrawals.
  - Enhanced the testing suite with new routines for performance testing.

- **Chores**
  - Updated Docker Compose configuration for the Solana service to include a stress profile and restart policy.
  - Increased SOL airdrop amounts in the Solana startup script.

### CODE REVIEW

File: cmd/zetae2e/local/local.go [hunk 1]
```diff
@@ -382,6 +383,24 @@ func localE2ETest(cmd *cobra.Command, _ []string) {
...
 		eg.Go(ethereumWithdrawPerformanceRoutine(conf, deployerRunner, verbose, e2etests.TestStressEtherWithdrawName))
+		eg.Go(
+			solanaDepositPerformanceRoutine(
+				conf,
+				deployerRunner,
+				verbose,
+				e2etests.TestStressSolanaDepositName,
+				e2etests.TestStressSPLDepositName,
+			),
+		)
```
Comment (gartnera) [inline cmd/zetae2e/local/local.go:394]: Would be nice if we could parallelize the SOL and SPL deposits. Running them serially seems to be what's slowing down the performance tests a lot.

File: e2e/e2etests/test_stress_solana_withdraw.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	r.Logger.Print("Latency report:")
+	r.Logger.Print("min:  %.2f", desc.Min)
+	r.Logger.Print("max:  %.2f", desc.Max)
+	r.Logger.Print("mean: %.2f", desc.Mean)
+	r.Logger.Print("std:  %.2f", desc.Std)
+	for _, p := range desc.DescriptionPercentiles {
+		r.Logger.Print("p%.0f:  %.2f", p.Percentile, p.Value)
+	}
```
Comment (gartnera) [inline e2e/e2etests/test_stress_solana_withdraw.go:92]: output from the logs for reference:
```
perf_sol_wit | Latency report:
perf_sol_wit | min:  19.55
perf_sol_wit | max:  23.12
perf_sol_wit | mean: 21.37
perf_sol_wit | std:  1.25
perf_sol_wit | p50:  20.59
perf_sol_wit | p75:  22.57
perf_sol_wit | p90:  23.06
perf_sol_wit | p95:  23.09
perf_sol_wit | all SOL withdrawals completed
perf_sol_wit | ✅ completed in 1m41.383067077s - stress test SOL withdrawals
```

File: changelog.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 * [3205](https://github.com/zeta-chain/node/issues/3205) - move Bitcoin revert address test to advanced group to avoid upgrade test failure
 * [3254](https://github.com/zeta-chain/node/pull/3254) - rename v2 E2E tests as evm tests and rename old evm tests as legacy
 * [3095](https://github.com/zeta-chain/node/pull/3095) - initialize simulation tests for custom zetachain modules
+* [3276](https://github.com/zeta-chain/node/pull/3276) - solana performance tests
```
Comment (lumtis) [inline changelog.md:14]:
```suggestion
* [3276](https://github.com/zeta-chain/node/pull/3276) - add Solana E2E performance tests
```

File: cmd/zetae2e/local/local.go [hunk 4]
```diff
@@ -382,6 +383,37 @@ func localE2ETest(cmd *cobra.Command, _ []string) {
...
+		eg.Go(
+			solanaWithdrawPerformanceRoutine(
+				conf,
+				"perf_sol_withdraw",
+				deployerRunner,
+				verbose,
+				conf.AdditionalAccounts.UserSolana,
+				e2etests.TestStressSolanaWithdrawName,
+				e2etests.TestStressSPLWithdrawName,
+			),
```
Comment (lumtis) [inline cmd/zetae2e/local/local.go:415]: Any reason we don't have SPL withdraw test? I see it is defined below

**Review by lumtis — COMMENTED** [review]
Looks good

Any way we can test with E2E CI tool?

**Review by ws4charlie — APPROVED** [review]
tests look good.

File: zetaclient/chains/solana/signer/signer.go [hunk 5]
```diff
@@ -170,26 +186,69 @@ func (signer *Signer) TryProcessOutbound(
...
+	for i := 0; i < broadcastRetries; i++ {
+		time.Sleep(backOff)
+
+		// PDA nonce may already be increased by other relayer, no need to retry
+		pdaInfo, err := signer.client.GetAccountInfo(ctx, signer.pda)
+		if err != nil {
+			logger.Error().Err(err).Fields(lf).Msgf("unable to get PDA account info")
+		} else {
+			pda := contracts.PdaInfo{}
+			err = borsh.Deserialize(&pda, pdaInfo.Bytes())
```
Comment (lumtis) [inline zetaclient/chains/solana/signer/signer.go:221]: Can we move this in the Solana package maybe? To not have cryptography details in this package

**Review by lumtis — APPROVED** [review]

**Pull request merged**